openedx/features/content_type_gating/models.py

```diff
@@ -57,36 +50,24 @@ class ContentTypeGatingConfig(StackedConfigurationModel):
     )
 
     @classmethod
-    def has_full_access_role_in_masquerade(cls, user, course_key, course_masquerade, student_masquerade,
-                                           user_partition):
+    def has_full_access_role_in_masquerade(cls, user, course_key, course_masquerade):
         """
         The roles of the masquerade user are used to determine whether the content gate displays.
         The gate will not appear if the masquerade user has any of the following roles:
         Staff, Instructor, Beta Tester, Forum Community TA, Forum Group Moderator, Forum Moderator, Forum Administrator
         """
-        if student_masquerade:
+        if is_masquerading_as_specific_student(user, course_key):
             # If a request is masquerading as a specific user, the user variable will represent the correct user.
             if user and user.id and has_staff_roles(user, course_key):
                 return True
-        elif user_partition:
```

Comment:
So, I think the thing that this was trying to support was making it so that if you masqueraded as a Verified Enrollment user (generically), it would give you the FBE Full Access role, effectively.

However, I think your change does the same thing by using the new helper functions that @stvstnfrd added (I'm assuming as part of the MFE masquerade effort).

So, looks good to me.

Reply:
I'm just not sure why it ever required a passed in UserPartition though (and then only checking the masquerade for that particular partition). CourseDurationLimitConfig was checking the verified enrollment masquerading before, just by directly checking the two partitions we care about.

Like, the masquerading is not affected by the passed-in partition. It is the same masquerade regardless. The incoming user partition didn't seem useful at all for checking the masquerade. I was just nervous I was missing part of the logic of the whole partition system.

Reply:
(To be clear, I understand the incoming partition was being used -- by checking if the masquerade matched it. But I just mean, that check seemed not useful to me?)

Reply:
But good that it seems to be a safe change!

Reply:
I think it was just that the masquerade functions that dealt w/ groups (before @stvstnfrd showed up) wanted a `user_partition`, and then pulled the id off the partition.